anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -820,26 +820,25 @@ def _switch_active_grypedb(
         inactive_records = self._find_match(db, active=False)
         for inactive_record in inactive_records.all():
             catalog_client.delete_document(
-                inactive_record.group_name, inactive_record.checksum
+                inactive_record.group_name, inactive_record.archive_checksum
             )
         inactive_records.delete(synchronize_session="evaluate")
         # search for active and mark inactive
         self._find_match(db, active=True).update(
-            {GrypeDBMetadata.active: False}, synchronize_session="evaluate"
+            {GrypeDBFeedMetadata.active: False}, synchronize_session="evaluate"
         )
```

Comment:
The use of synchronization strategy is not correct here. Evaluate as a synchronization strategy means locate objects with in the session but the intent here is to lookup all the active/inactive db records.  https://docs.sqlalchemy.org/en/14/orm/session_basics.html#selecting-a-synchronization-strategy sqlalchemy generally recommends refraining from using the sync strategy

The recommended way of operating on db objects is by directly updating them (sqlalchemy recognizes the changes and updates the record when the session is committed) or by using the session to delete the object. As I indicated above find_match should return the results, not the query (the naming is confusing here)

```
        inactive_records = self._find_match(db, active=False)
        for inactive_record in inactive_records:
            catalog_client.delete_document(
                inactive_record.group_name, inactive_record.archive_checksum
            )
           db.delete(inactive_record)

       for active_record in self._find_match(db, active=True):
          active_record.active = False
```

Reply:
thanks! i did the sync strategy stuff when i was still relatively new to sqlalchemy. docs definitely make more sense now than they did then. will fix. (i think zane might do this since he's working on that method in this branch right now)